.github/actions/templates/validateModuleDeployment/action.yml

```diff
@@ -84,14 +87,17 @@ runs:
             @{ Name = 'managementGroupId'; Value = '${{ inputs.managementGroupId }}' }
             @{ Name = "tenantId"; Value = '${{ env.ARM_TENANT_ID }}' }
             @{ Name = "deploymentSpId"; Value = '${{ env.DEPLOYMENT_SP_ID }}' }
-        )
+        ) | ForEach-Object { [PSCustomObject]$PSItem }
 
-        $DefaultParameterFileTokens = $DefaultParameterFileTokens | ForEach-Object { [PSCustomObject]$PSItem }
+        # Get additional Custom Parameter File Tokens from input
+        Write-Verbose 'Additional Custom Parameter File Tokens: ${{ inputs.customParameterFileTokens }}' -Verbose
+        $OtherCustomParameterFileTokens = '${{  inputs.customParameterFileTokens }}' | ConvertFrom-Json
 
         # Construct Token Function Input
         $ConvertTokensInputs = @{
             ParameterFilePath                 = '${{ inputs.parameterFilePath }}'
             DefaultParameterFileTokens        = $DefaultParameterFileTokens
+            OtherCustomParameterFileTokens    = $OtherCustomParameterFileTokens
             LocalCustomParameterFileTokens    = $Settings.parameterFileTokens.localTokens.tokens
```

Comment:
Does it maybe make sense to somewhat merge the extra token into the 'local' or 'default' tokens and hence avoid an additional parameter?

Reply:
For how the token replacement function is implemented:

- `DefaultParameterFileTokens`: take care of the secrets replacement
- `LocalCustomParameterFileTokens`: take care of local tokens added in the settings file
- `OtherCustomParameterFileTokens`: take care of additional tokens, hence I thought about leveraging that existing parameter

At this point I'm not against an update of the function itself to get only one `ParameterFileTokens`, maybe setting it as mandatory (they're currently all optional). Another update I'd suggest would be to set its type to hashtable, a list of key values since tokens anyway need to be unique. I'd vote for discussing those changes outside of this PR.

Reply:
Agreed. I was under the impression the parameter did not exist (shame on me)